aiida/orm/nodes/process/process.py

```diff
@@ -104,17 +103,7 @@ def set_process_type(self, process_class):
 
         :param process_class: the process class using this process node as storage
         """
-        class_module = process_class.__module__
-        class_name = process_class.__name__
-
-        # If the process is a registered plugin the corresponding entry point will be used as process type
-        process_type = get_entry_point_string_from_class(class_module, class_name)
-
-        # If no entry point was found, default to fully qualified path name
-        if process_type is None:
-            self.process_type = '{}.{}'.format(class_module, class_name)
-        else:
-            self.process_type = process_type
+        self.process_type = process_class.build_process_type()
```

Comment:
I would instead delegate the building of the process type itself to the `Process` instance that actually calls `set_process_type`. The setter here then simply takes the value it receives as is and assigns it (with optional validation but don't think we had any or need any now)